Include AppStream metadata

aleixpol opened the pull request:
## Description
Provide proper metadata to have QGIS listed in software centers such as Gnome Software and KDE Discover.
You can learn more about appstream here: https://www.freedesktop.org/wiki/Distributions/AppStream/

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

aleixpol committed:
Install appdata information for QGIS

jef-n commented on CMakeLists.txt:
```diff
@@ -906,3 +906,7 @@ IF (WITH_CORE)
   INCLUDE(CPack)
 ENDIF (WITH_CORE)
+
+IF (LINUX)
```
`LINUX` is not defined - `UNIX AND NOT APPLE` might work - there already is a path for this (in line 637)

jef-n commented on linux/CMakeLists.txt:
```diff
@@ -0,0 +1,7 @@
+install(FILES org.qgis.qgis.desktop DESTINATION share/applications)
+install(FILES org.qgis.qgis.appdata.xml DESTINATION share/metainfo)
+install(FILES icons/qgis_icon.svg DESTINATION share/icons/hicolor/scalable/apps)
```
`qgis_icon.svg` is not in `linux` but `../images` (maybe all this should be moved there instead of `linux/`).  Doesn't work for me.
Also `qgis_icon.svg` is expected to be installed to `/usr/share/pixmaps` (see the original `debian/rules`; otherwise `debian/qgis.install` needs to be adapted too; also applies to other images and other directories).

jef-n commented on linux/CMakeLists.txt:
```diff
@@ -0,0 +1,7 @@
+install(FILES org.qgis.qgis.appdata.xml DESTINATION share/metainfo)
+install(FILES icons/qgis_icon.svg DESTINATION share/icons/hicolor/scalable/apps)
+
+foreach(_type IN qgs qlr qml qpt)
```
No `IN` needed - doesn't work for me like this.

jef-n reviewed (commented):
Not sure about the rpm part - the debian part doesn't work.

aleixpol commented:
> Why all the changes to the rules file though? Are they required for appstream or an unrelated change?

Ah sorry, I completely forgot to mention it. I was working on a flatpak build of QGIS, and to have it properly listed one needs appstream information available.
https://github.com/aleixpol/qgis-flatpak

I did the change so that we install properly as we do on other cmake-based projects by just installing from cmake rather than having every packaging format doing it ad-hoc. You can see I broke it down in few separate commits.

nyalldawson commented:
For speed of review I'd split these two pieces of work into separate PRs - so one which is ONLY the appstream info addition, and then a second one for the packaging changes.

nyalldawson commented:
Thanks!

Looking at the spec, here's my thoughts:

- I think we need the `launchable` tag
- We need the compulsory 'releases' tag: "The application metainfo should at least provide one <releases/> tag, which has one or more <release/> childs to define the version and release date of this application. "
- Could we add some screenshots? E.g. some from https://qgis.org/en/site/about/screenshots.html
- We should add a project_group tag for "QGIS.org"

aleixpol commented:
> I think we need the launchable tag

Added

> We need the compulsory 'releases' tag: "The application metainfo should at least provide one tag, which has one or more childs to define the version and release date of this application. "

Added, although despite being mandatory this still validates without. Will have to remember to keep it up to date.

> Could we add some screenshots? E.g. some from https://qgis.org/en/site/about/screenshots.html

If you provide me the url to fetch it from, I'm not sure how to get it. Is it even allowed to point at flickr?

> We should add a project_group tag for "QGIS.org"

Added

nyalldawson commented:
Yep - screenshots looks complex. Let's leave that for a follow up later.

nyalldawson commented:
I'm +1 to merge, but a second pair of eyes would be good

m-kuhn commented:
Good to see movement on the packaging front. It looks good to me although I'm not an expert on the topic. I would vote to merge and incrementally address potential issues. :+1:

nyalldawson commented:
1 to go!
